src/reconcile/rec_write.c

```diff
@@ -2401,8 +2405,12 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         WT_ASSERT_ALWAYS(session, chunk->entries > 0, "Trying to write an empty chunk");
     }
 
-    if (last_block)
-        WT_RET(__rec_build_delta(session, r));
+    if (last_block) {
+        WT_RET(__rec_build_delta(session, r, &build_delta));
+        /* Discard the delta if it is larger than one tenth of the size of the full image. */
+        if (build_delta && r->delta.size > chunk->image.size / 10)
+            build_delta = false;
```

Comment:
I'm assuming the delta will get cleaned up at some later point if we're not using it?

Reply:
yes, when the reconcile structure is destroyed.